drivers/stmpe811/stmpe811.c

```diff
@@ -262,12 +262,12 @@ int stmpe811_read_touch_state(const stmpe811_t *dev, stmpe811_touch_state_t *sta
         return -STMPE811_ERR_I2C;
     }
 
+    _clear_interrupt_status(dev);
+
     if ((val & STMPE811_TSC_CTRL_STA)) {
-        _clear_interrupt_status(dev);
         *state = STMPE811_TOUCH_STATE_PRESSED;
     }
     else {
-        _reset_fifo(dev);
```

Comment:
This is required otherwise on the next press you will get the last pressed position first.